spec.html

```diff
@@ -28857,10 +28857,18 @@ <h1>
         </dl>
         <emu-alg>
           1. Assert: If _direct_ is *false*, then _strictCaller_ is also *false*.
-          1. If _x_ is not a String, return _x_.
+          1. If _x_ is an Object, then
+            1. Let _code_ be HostGetCodeForEval(_x_).
+            1. If _code_ is ~unknown~ return _x_.
+            1. Else, let _xStr_ be _code_.
+          1. Else if _x_ is a String, then
+            1. Let _xStr_ be _x_.
+          1. Else,
+            1. Return _x_.
```

Comment:
Ordering nit:
```suggestion
          1. If _x_ is a String, then
            1. Let _xStr_ be _x_.
          1. Else if _x_ is an Object, then
            1. Let _code_ be HostGetCodeForEval(_x_).
            1. If _code_ is a String, let _xStr_ be _code_.
            1. Else, return _x_.
          1. Else,
            1. Return _x_.
```
Also possible:
```suggestion
          1. If _x_ is an Object, let _xStr_ be HostGetCodeForEval(_x_). Otherwise, let _xStr_ be _x_.
          1. If _xStr_ is not a String, return _x_.
```

Reply:
Note also that a later use of _x_ needs updating:
```diff
-1. Let _script_ be ParseText(StringToCodePoints(_x_), |Script|).
+1. Let _script_ be ParseText(StringToCodePoints(_xStr_), |Script|).
```